pkg/endpoint/metadata/k8s_metadatafetcher.go

```diff
@@ -91,26 +97,41 @@ func (cemf *cachedEndpointMetadataFetcher) getPod(namespace, name string) (*slim
 	_, initWatch := cemf.pods.Initialized(cemf.db.ReadTxn())
 	<-initWatch
```

Comment:
Also, it looks that we are exposing the internal details of the statedb? Why doesn't this `initWatch` also means that pods are ready to be "fetched"?

Reply:
I meant the internal details with the `pod, _, watch, found := cemf.pods.GetWatch`.

In a sense that, why can't we always return the `GetWatch` "fully synced"? Why does the caller needs to know the internals of `GetWatch` so that it needs to understand that it needs to wait for `watch` to be closed? Is there a use case where the caller does `cemf.pods.GetWatch` without waiting for watch to be closed?